File: ios/VideoPreviewViewModule.swift
import ExpoModulesCore

public class VideoPreviewViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("VideoPreviewViewModule")

        View(VideoPreviewView.self) {
            Prop("captureDeviceId") { (view: VideoPreviewView, prop: String) in
                view.captureDeviceId = prop
            }

            Prop("mirrorVideo") { (view: VideoPreviewView, prop: Bool) in
                view.mirrorVideo = prop
            }

            Prop("videoLayout") { (view: VideoPreviewView, prop: String) in
                view.videoLayout = prop
            }
        }
    }
}

File: android/src/main/java/org/membraneframework/reactnative/VideoPreviewViewModule.kt
package org.membraneframework.reactnative

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class VideoPreviewViewModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("VideoPreviewViewModule")

        View(VideoPreviewView::class) {
            Prop("videoLayout") { view: VideoPreviewView, videoLayout: String ->
                view.setVideoLayout(videoLayout)
            }

            Prop("mirrorVideo") { view: VideoPreviewView, mirrorVideo: Boolean ->
                view.setMirrorVideo(mirrorVideo)
            }

            Prop("captureDeviceId") { view: VideoPreviewView, captureDeviceId: String ->
                view.switchCamera(captureDeviceId)
            }
        }
    }
}